ome/resource_discovery.go

```diff
@@ -48,6 +53,74 @@ func (r *discoveryResource) Schema(_ context.Context, _ resource.SchemaRequest,
 	}
 }
 
+func (r *discoveryResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	var data models.OmeDiscoveryJob
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if data.Schedule.ValueString() == "RunNow" {
+		if !data.Cron.IsNull() {
+			resp.Diagnostics.AddAttributeError(
+				path.Root("cron"),
+				"Attribute Error",
+				"With Schedule as RunNow, CRON can't be set.",
+			)
+		}
+	}
+
+	if len(data.DiscoveryConfigTargets) > 0 {
+		for idx, dct := range data.DiscoveryConfigTargets {
+			idxError := "Inappropriate value for attribute `discovery_config_targets`: element: " + strconv.Itoa(idx) + "\n"
+			if dct.Redfish == nil && dct.SNMP == nil && dct.SSH == nil && dct.WSMAN == nil {
+				resp.Diagnostics.AddAttributeError(
+					path.Root("discovery_config_targets").AtListIndex(idx),
+					"Attribute Error",
+					idxError+"Atleast one of protocol should be configured for the discovery targets.")
+			}
+			if len(dct.DeviceType) > 0 {
+				for idx, dt := range dct.DeviceType {
+					currDT := dt.ValueString()
+					if !isDeviceType(currDT) {
+						resp.Diagnostics.AddAttributeError(
+							path.Root("discovery_config_targets").AtListIndex(idx).AtName("device_type"),
+							"Attribute Error",
+							idxError+"The device type list should contain the following values: `SERVER`, `CHASSIS`, `NETWORK SWITCH`, and `STORAGE`.")
+					}
+				}
+			} else {
+				resp.Diagnostics.AddAttributeError(
+					path.Root("discovery_config_targets").AtListIndex(idx).AtName("device_type"),
+					"Attribute Error",
+					idxError+"Atleast one of device type should be configured. ")
+			}
+			if len(dct.NetworkAddressDetail) == 0 {
+				resp.Diagnostics.AddAttributeError(
+					path.Root("discovery_config_targets").AtListIndex(idx).AtName("network_address_detail"),
+					"Attribute Error",
+					idxError+"Atleast one of network address detail should be configured. ")
+			}
+		}
+	} else {
```

Comment:
These length validations can be done using https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators@v0.11.0/listvalidator#SizeAtLeast, rt?

Reply:
added the list validators.